vm-setup/roles/common/defaults/main.yml

```
@@ -79,4 +106,11 @@ external_network:
           addr: "{% if external_network_cidr_v4|ipv4 != False %}127.0.0.1{% else %}::1{% endif %}"
       srvs: "{{dns_externalsrvs | default([])}}"
 
-networks: "{{ provisioning_network + external_network }}"
+# Provisioning network is bridged and external network is nated
+networks_mixed: "{{ provisioning_network_bridge + external_network }}"
+# Both networks are nated
+networks_nated: "{{ provisioning_network_nat + external_network }}"
+# Enable only nated networks
+networks_nat_only: "{{ lookup('env', 'ENABLE_NATED_PROVISIONING_NETWORK') | bool }}"
+# Placeholder for the eventually selected network composition
+networks: ""
```

If we do it like this instead we don't need the `set_fact` tasks.
```suggestion
networks: "{{ networks_nat_only | ternary(networks_nated, networks_mixed) }}"
```

---

I am not sure I like these very abstracted  Ansible filters... for me it is more in line with the sequential execution of the tasks  to set these values when they are needed  but I see your point but I personally don't like it (no technical issue just personal preference).

---

I see. I find it much easier to understand what is going on when setting it like this directly instead of having to hunt down the `set_fact`s. The ternary operator is not very abstract even, it is just uncommon to spell it out like they do in ansible. Another name for it is "inline if" and a [more common way](https://en.wikipedia.org/wiki/Ternary_conditional_operator) to express it would be something like `a ? b : c`.

Would it be better if we use actual if/else inline? Like this:

```suggestion
networks: "{{ networks_nated if networks_nat_only else networks_mixed }}"
```

---

Okay, I can settle for the regular inline if that is fine for me too.